server.js

```diff
@@ -14,20 +14,9 @@ export function injectRSCPayload(rscStream, options) {
   let buffered = [];
   let timeout = null;
   function flushBufferedChunks(controller) {
-    for (let chunk of buffered) {
-      let buf = decoder.decode(chunk, {stream: true});
-      if (buf.endsWith(trailer)) {
-        buf = buf.slice(0, -trailer.length);
-      }
-      controller.enqueue(encoder.encode(buf));
-    }
-
-    let remaining = decoder.decode();
-    if (remaining.length) {
-      if (remaining.endsWith(trailer)) {
-        remaining = remaining.slice(0, -trailer.length);
-      }
-      controller.enqueue(encoder.encode(remaining));
+    for (const chunk of buffered) {
+      const html = decoder.decode(chunk, { stream: true });
+      controller.enqueue(encoder.encode(html));
```

Comment:
Perhaps, trailer can be still stripped off by `html.endsWith(trailer)` check?

Or is decode/encode unnecessary if trailer is stripped off as binary like this? :thinking: 

```js
const trailerBin = new TextEncoder().encode('</body></html>').reverse()

    for (let chunk of buffered) {
      if (trailerBinReverse.every((byte, i) => chunk[chunk.length - i - 1] === byte)) {
        chunk = chunk.slice(0, -trailerBinReverse.length)
      }
      controller.enqueue(chunk)
```